test/box-luatest/gh_7113_rev_tree_iters_gap_tracking_test.lua

```diff
@@ -0,0 +1,136 @@
+                                       {2, 'unsigned'}}})
+        s:insert{0, 0}
+        s:insert{1, 0}
+    end)
+end)
+
+g.after_each(function()
+    g.server:eval('box.space.s:drop()')
+end)
+
+g['test_reverse_iter_gap_tracking'] = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local txn_proxy = require('test.box.lua.txn_proxy')
+
+        local tx = txn_proxy:new()
+
+        local conflict_err = 'Transaction has been aborted by conflict'
+
+        tx:begin()
+        tx('box.space.s:select({1, 0}, {iterator = "LT"})')
+        box.space.s:insert{0, 1}
+        tx('box.space.s:insert{2, 0}')
+        t.assert_equals(tx:commit(), {{error = conflict_err}})
+        box.space.s:delete{0, 1}
+
+        tx:begin()
+        tx('box.space.s:select({1, 0}, {iterator = "LE"})')
+        box.space.s:insert{0, 1}
+        tx('box.space.s:insert{2, 0}')
+        t.assert_equals(tx:commit(), {{error = conflict_err}})
+        box.space.s:delete{0, 1}
+
+        tx:begin()
+        tx('box.space.s:select({1}, {iterator = "LT"})')
+        box.space.s:insert{0, 1}
+        tx('box.space.s:insert{2, 0}')
+        t.assert_equals(tx:commit(), {{error = conflict_err}})
+        box.space.s:delete{0, 1}
+
+        tx:begin()
+        tx('box.space.s:select({1}, {iterator = "LE"})')
+        box.space.s:insert{0, 1}
+        tx('box.space.s:insert{2, 0}')
+        t.assert_equals(tx:commit(), {{error = conflict_err}})
+        box.space.s:delete{0, 1}
+
+        tx:begin()
+        tx('box.space.s:select({0}, {iterator = "REQ"})')
+        box.space.s:insert{0, 1}
+        tx('box.space.s:insert{2, 0}')
+        t.assert_equals(tx:commit(), {{error = conflict_err}})
+        box.space.s:delete{0, 1}
+
+        tx:begin()
+        tx('box.space.s:select({1}, {iterator = "REQ"})')
+        box.space.s:insert{1, 1}
+        tx('box.space.s:insert{2, 0}')
+        t.assert_equals(tx:commit(), {{error = conflict_err}})
+        box.space.s:delete{1, 1}
+
+        tx:begin()
+    end)
+end
+
+g['test_reverse_iter_clarify_before_gap_tracking'] = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local txn_proxy = require('test.box.lua.txn_proxy')
+
+        local tx = txn_proxy:new()
+
+        --[[
+            The following tests are a safety net for catching the buggy case
+            when tuple clarification could be done after gap tracking
+            (gh-7073).
+        --]]
+        box.internal.memtx_tx_gc(128)
+
+        tx:begin()
+        box.space.s:delete{0, 0}
+        t.assert_equals(tx("box.space.s:select({1, 0}, {iterator = 'LT'})"),
+                        {{}})
+        tx:commit()
+        box.space.s:insert{0, 0}
+
+        tx:begin()
+        box.space.s:delete{0, 0}
+        t.assert_equals(tx("box.space.s:select({0, 0}, {iterator = 'LE'})"),
+                        {{}})
+        tx:commit()
+        box.space.s:insert{0, 0}
```

Comment:
Could you please explain what this test may catch? (I've run this snippet for 100000 times on master and nothing happened)